docs/storage-design.md

```diff
@@ -0,0 +1,149 @@
+
+A Document has the following properties:
+* ExternalID: the identifier for the Document in the actual storage system being used
+* CreatedBy: The user that uploaded the Document
+* CreatedOn: When it was uploaded
+* DisplayName
+* Description
+* Tags (to allow filtering)
+* Size
+* Mime type (required so that requests know how to handle it)
+* ...
+
+Other properties that can be added to expand the capabilities of storage / security around storage include:
+* Encrypted at rest or not
+* Uploading source IP
+* ...
+
+Important to note is that the platform in essence adds a bunch of meta data around the underlying storage of the file. 
+
+### Accessing the Files
+The files will be accessed via Rest requests.
+
+The Rest api for documents will then check the request matches the 
+
+### Usage
+The following entities in the Alkemio domain model are expected to have a Storage Bucket:
+* Hub
+* Challenge: essentially a child of the parent Space Storage Bucket.
+* Platform
+* Library
+* Organization
+
+In the first version of Storage Bucket usage there are no quotas on files that are uploaded, but the 
+Note that Platform and Library are essentially non-managed storage Buckets; we can look later if we need 
+to put restrictions around content that is stored there.  
+
+**Note**: as part of this we would allow all Registered users to be able to upload files on their Profile, which then 
+uses the Platform StorageBucket. This is somewhat of a risk (and the reason we did not support this initially) in terms of providing unlimited resource accessibility, but a key 
+difference is that we now have an audit trail of who uploaded each file - so we can also easily aggregate to see how much content each user is uploading if so desired.
+
+### Knowing the StorageBucket to use ==> Profile
+The Storage Bucket to use is tied to each Profile entity. This is chosen as that is where visuals are uploaded, and also references are essentially how we currently work with attachments.
+
+This does mean that usage of image uploading or adding files are references depends on Profile entity - which means we should only use Visual entities as part of a Profile. This has implications for e.g. InnovationBucket usage with its Branding. It maybe this needs to use a Profile internally and expose the visual logically via Branding entity on the api.
+
+## Known issues
+* Deletion of files: a side effect of content addressed storage is that if two users upload the same document then it will get the same externalID. 
+  * This raises the potential issue of two users uploading the same Document, and one deletes it. If we immediately deleted the file on IPFS then the second user's document contents would no longer be accessible
+  * For now we do *not* delete the file directly on IPFS if a Document is deleted, but would need to add in a check if another Document has the same externalID or not before removing it. Once we have ensured there is no direct access to 
```

Comment:
We don't have delete API at all at the moment

Reply:
We do have the ability to delete a document: 
![image](https://github.com/alkem-io/alkemio/assets/30729240/5db15808-58ea-4107-8d58-49e0a026ffb2)

But that does not at the moment actually deleted in the underlying store.

Reply:
Ok, I stand corrected. Our delete is an actual IPFS unpin, hence the confusion...

Reply:
I think in the current setup we must remove the mutation actually, it is borderline dangerous... We don't want people to unpin a CID used in another document, potentially.